Unregister receiver crash for Pre-Lollipop devices

Got a crash on an Asus Transformer (API 16) when rotating the device. I unsubscribe in `onPause` and subscribe in `onResume` but I think there's a problem unregistering the receiver if it's not attached.

I have a subscription from it to a `BehaviourSubject` and multiple subscriptions from the `BehaviourSubject` to other subscribers.

```
09-29 11:04:31.350 1174-1174/uk.co.imagitech.learn2.hp E/ActivityThread: Activity uk.co.imagitech.learn2.hp.MainActivity has leaked IntentReceiver com.github.pwittchen.reactivenetwork.library.network.observing.strategy.PreLollipopNetworkObservingStrategy$1$1@416c8ad0 that was originally registered here. Are you missing a call to unregisterReceiver()?
     android.app.IntentReceiverLeaked: Activity uk.co.imagitech.learn2.hp.MainActivity has leaked IntentReceiver com.github.pwittchen.reactivenetwork.library.network.observing.strategy.PreLollipopNetworkObservingStrategy$1$1@416c8ad0 that was originally registered here. Are you missing a call to unregisterReceiver()?
         at android.app.LoadedApk$ReceiverDispatcher.<init>(LoadedApk.java:792)
         at android.app.LoadedApk.getReceiverDispatcher(LoadedApk.java:593)
         at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1120)
         at android.app.ContextImpl.registerReceiver(ContextImpl.java:1107)
         at android.app.ContextImpl.registerReceiver(ContextImpl.java:1101)
         at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:365)
         at com.github.pwittchen.reactivenetwork.library.network.observing.strategy.PreLollipopNetworkObservingStrategy$1.call(PreLollipopNetworkObservingStrategy.java:50)
         at com.github.pwittchen.reactivenetwork.library.network.observing.strategy.PreLollipopNetworkObservingStrategy$1.call(PreLollipopNetworkObservingStrategy.java:42)
         at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
         at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
         at rx.Observable.unsafeSubscribe(Observable.java:10150)
         at rx.internal.operators.OperatorSubscribeOn$1.call(OperatorSubscribeOn.java:94)
         at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:228)
         at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:442)
         at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:305)
         at java.util.concurrent.FutureTask.run(FutureTask.java:137)
         at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:150)
         at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:264)
         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
         at java.lang.Thread.run(Thread.java:856)
09-29 11:04:31.360 1174-1174/uk.co.imagitech.learn2.hp E/AndroidRuntime: FATAL EXCEPTION: main
     java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
         at rx.android.schedulers.LooperScheduler$ScheduledAction.run(LooperScheduler.java:114)
         at android.os.Handler.handleCallback(Handler.java:615)
         at android.os.Handler.dispatchMessage(Handler.java:92)
         at android.os.Looper.loop(Looper.java:137)
         at android.app.ActivityThread.main(ActivityThread.java:4745)
         at java.lang.reflect.Method.invokeNative(Native Method)
         at java.lang.reflect.Method.invoke(Method.java:511)
         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
         at dalvik.system.NativeStart.main(Native Method)
      Caused by: java.lang.IllegalArgumentException: Receiver not registered: com.github.pwittchen.reactivenetwork.library.network.observing.strategy.PreLollipopNetworkObservingStrategy$1$1@416c8ad0
         at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:654)
         at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1143)
         at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:378)
         at com.github.pwittchen.reactivenetwork.library.network.observing.strategy.PreLollipopNetworkObservingStrategy$1$2.call(PreLollipopNetworkObservingStrategy.java:54)
         at com.github.pwittchen.reactivenetwork.library.network.observing.strategy.PreLollipopNetworkObservingStrategy$2$1.call(PreLollipopNetworkObservingStrategy.java:71)
         at rx.android.schedulers.LooperScheduler$ScheduledAction.run(LooperScheduler.java:107)
         at android.os.Handler.handleCallback(Handler.java:615) 
         at android.os.Handler.dispatchMessage(Handler.java:92) 
         at android.os.Looper.loop(Looper.java:137) 
         at android.app.ActivityThread.main(ActivityThread.java:4745) 
         at java.lang.reflect.Method.invokeNative(Native Method) 
         at java.lang.reflect.Method.invoke(Method.java:511) 
         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
         at dalvik.system.NativeStart.main(Native Method) 
```

--- a/library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java
+++ b/library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java
@@ -51,7 +51,12 @@ public class PreLollipopNetworkObservingStrategy implements NetworkObservingStra
 
         subscriber.add(unsubscribeInUiThread(new Action0() {
           @Override public void call() {
-            context.unregisterReceiver(receiver);
+            //Catch case where the receiver is already unregistered
+            try {
+              context.unregisterReceiver(receiver);
+            } catch (Exception e) {
+              // ignore exception
+            }
           }
         }));
       }

